### PR TITLE
Fixed #2650: Redirect to the valid commit

### LINKED ISSUE #2650: Clicking on Version v2.4.0 leads to an undefined commit on https://telescope.cdot.systems/
**What happened**:

When you select the version, it's redirects to an undefined (404) commit

image

**What should have happened**:

When you select the version, it's supposed to take you to the GitHub commit relating to that release version

### PR DESCRIPTION
**Pull request by tcvan0707**
## Issue This PR Addresses
Fixes #2650 

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

When we click on the version number, it would redirect us to the valid commit related to the release.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — COMMENTED** [review]
Can you explain this change a bit?

**Comment by tcvan0707** [comment]
> Can you explain this change a bit?

Right now, when we click on the version, for example `v2.4.0`, it would get us to the 404 page, and this is not what we expected when we push a commit. And the change I have made, including the `head_commit` to redirect us to the commit related to the release when the `push` event happens.

File: tools/autodeployment/webhook.js [hunk 1]
```diff
@@ -44,7 +45,8 @@ function requestFilter(repo, buildType, action, ref, mainBranch) {
     // Check if the reference is the branch used as main or if it is a tag
     (currentBranch === mainBranch || ref.includes('tags/')) &&
     // Check the build type and the action
-    (buildType === 'staging' || (buildType === 'production' && action === 'published'))
+    (buildType === 'staging' || (buildType === 'production' && action === 'published')) &&
+    headCommit === 'master'
```
Comment (humphd) [inline tools/autodeployment/webhook.js:49]: I don't understand this.  The `head_commit` is going to be an object that looks like this (see https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-2):

```json
  "head_commit": {
    "id": "6113728f27ae82c7b1a177c8d03f9e96e0adf246",
    "tree_id": "4b825dc642cb6eb9a060e54bf8d69288fbee4904",
    "distinct": true,
    "message": "Adding a .gitignore file",
    "timestamp": "2019-05-15T15:20:41Z",
    "url": "https://github.com/Codertocat/Hello-World/commit/6113728f27ae82c7b1a177c8d03f9e96e0adf246",
    "author": {
      "name": "Codertocat",
      "email": "21031067+Codertocat@users.noreply.github.com",
      "username": "Codertocat"
    }
```

**Review by humphd — CHANGES_REQUESTED** [review]

File: tools/autodeployment/webhook.js (hunk 1 above)
Comment (AmasiaNalbandian) [reply tools/autodeployment/webhook.js:49]: @tcvan0707 Can we look into what @manekenpix suggested on the issue, to use `target_commitish` from the [webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-36) from GitHub? The payload we get to grab the `target_commitish` is found on [this line](https://github.com/Seneca-CDOT/telescope/blob/cfc8f8f30f67869715f3dceed836cd76c4c76275/tools/autodeployment/webhook.js#L66)

Comment (AmasiaNalbandian) [reply tools/autodeployment/webhook.js:49]: We can move the const reference to be global so you can grab reference outside of the function(but make sure its valid before we use it). We can try a build with this to test it, let us know when you are ready!

**Pull request closed without merging**